addons/l10n_hu_edi/models/l10n_hu_edi_transaction.py

```diff
@@ -0,0 +1,531 @@
+                if t['username'] == self.credentials_username
+                   and t['source'] == 'MGM'
+            ]
+
+            for transaction_code in transaction_codes_to_query:
+                try:
+                    invoices_results = self.env['l10n_hu_edi.connection'].do_query_transaction_status(
+                        self.credentials_id.sudo(),
+                        transaction_code,
+                        return_original_request=True,
+                    )
+                except L10nHuEdiConnectionError as e:
+                    return self.write({
+                        'messages': {
+                            'error_title': _('Error querying active transactions while attempting timeout recovery.'),
+                            'errors': e.errors,
+                        },
+                    })
+
+                for invoice_result in invoices_results:
+                    invoice_name = invoice_result['original_invoice_xml'].findtext('{*}invoiceNumber')
+                    matched_transaction = self.filtered(lambda t: t.invoice_id.name == invoice_name)
+
+                    if matched_transaction:
+                        # Set the correct transaction code on the matched transaction
+                        matched_transaction.transaction_code = transaction_code
+                        matched_transaction._process_query_transaction_result(invoice_result)
+
+            available_pages = transaction_list['available_pages']
+            page += 1
+
+        # Any transactions that could not be matched to the query results should be regarded as not received by NAV.
+        self.filtered(lambda t: t.state == 'send_timeout').write({
+            'state': 'to_send',
+            'messages': {
+                'error_title': _('Sending failed due to time-out.'),
+                'errors': [],
+            }
+        })
+
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_production(self):
+        if (True, 'production') in self.mapped(lambda t: (t.is_active, t.credentials_mode)):
+            raise UserError(_('Cannot delete active transactions in production mode!'))
+
+    # === Actions === #
+
+    def action_download_file(self):
+        """ Download the XML file linked to the document.
+
+        :return: An action to download the attachment.
+        """
+        self.ensure_one()
+        return {
+            'type': 'ir.actions.act_url',
+            'url': f'/web/content/l10n_hu_edi.transaction/{self.id}/attachment_file/{self._get_xml_file_name()}?download=true',
+        }
+
+    def action_process(self):
+        self.ensure_one()
+        if self.state in ['to_send', 'token_error']:
+            self.upload()
+        elif self.state in ['sent', 'query_error']:
+            self.query_status()
+        elif self.state == 'send_timeout':
+            self.recover_timeout()
+
+    # === Helpers === #
+
+    @contextmanager
+    def _acquire_lock(self, no_commit=False):
```

Comment:
should be in account_move_send

Reply:
I don't see a locking of the moves in account_move_send (I see it in the account_move_send cron, but not in the wizard)